src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1,79 +1,1419 @@
+    const adminFillColors: string[] = ['black', 'red', 'orange', 'yellow', 'green'];
+    for (let t = 1; t < tilesets.length; t += 1) {
+      const adminFillLayer = {
+        ...fillLayerConfig,
+        id: `${ADMN0_EN}-admin-${t}-fill`,
+        paint: {
+          'fill-color': adminFillColors[t],
+          'fill-opacity': 0.75,
+        },
+        source: {
+          layer: tilesets[t].layer,
+          type: 'vector',
+          url: tilesets[t].url,
+        },
+        visible: t === 1,
+      };
+
+      const handler: Handlers = {
+        layer: [`${ADMN0_EN}-admin-${t}-fill`],
+        method: e => {
+          this.onAdminFillClick(e, country, t);
+        },
+        name: `${ADMN0_EN}-admin-${t}-fill-drilldown`,
+        type: 'click',
+      };
+      ADMIN_FILL_HANDLERS.push(handler);
+
+      if (t === 1) {
+        const adminFilterExpression: any[] = ['any'];
+        for (const jurisdiction of filteredJurisdictions) {
+          if (jurisdiction.geographic_level === 1) {
+            const filterExpression = [
+              '==',
+              tilesets[t].idField,
+              jurisdiction.name as string,
+            ].filter(e => !!e);
+            if (filterExpression.length === 3) {
+              adminFilterExpression.push(filterExpression);
+            }
+          }
+        }
+        if (adminFilterExpression.length > 1) {
+          (adminFillLayer as any).filter = [...adminFilterExpression];
+        }
+      }
+      ADMIN_FILL_LAYERS.push(adminFillLayer);
+    }
+
+    function getJurisdictionFillLayers(jurisdictions: Jurisdiction[]) {
+      const layers: FlexObject[] = [];
+      const geoGraphicLevels: number[] = [];
+
+      for (const j of jurisdictions) {
+        const { geographic_level } = j;
+        if (
+          typeof geographic_level !== 'undefined' &&
+          geoGraphicLevels.indexOf(geographic_level) === -1
+        ) {
+          geoGraphicLevels.push(geographic_level);
+        }
+      }
+      geoGraphicLevels.sort().reverse();
+      const colorMap = ['red', 'orange', 'yellow', 'green', 'blue', 'purple'];
+
+      for (const g of geoGraphicLevels) {
+        const featureCollection = {
+          features: jurisdictions.filter(j => j.geographic_level === g).map(j => j.geojson),
+          type: 'FeatureCollection',
+        };
+
+        const layer = {
+          ...fillLayerConfig,
+          id: `${ADMN0_EN}-admin-${g}-jurisdiction-fill`,
+          paint: {
+            'fill-color': colorMap[g],
+          },
+          source: {
+            ...fillLayerConfig.source,
+            data: {
+              ...fillLayerConfig.source.data,
+              data: JSON.stringify(featureCollection),
+            },
+          },
+          visible: true,
+        };
+        layers.push(layer);
+      }
+      return layers;
+    }
+
+    const JURISDICTION_FILL_LAYERS = getJurisdictionFillLayers(filteredJurisdictions);
+
+    const drillUpHandler: Handlers = {
```

Comment:
Should we move this to the module that has the other Gisida handlers?

Reply:
Not sure since these handlers call private functions of the IRS plan page  which are very specific to the interaction of drilling up and down through the Jurisdiction hierarchies. If we extract the map itself from this page then maybe but then we would probably have to migrate some of this internal component state into redux stores for the separate modules to work together like this.

Reply:
Maybe that is worthy of another issue?

Reply:
Issue created hapa: #243